src/vm/crossloaderallocatorhash.inl

```diff
@@ -1099,7 +1099,23 @@ LAHASHDEPENDENTHASHTRACKERREF CrossLoaderAllocatorHash<TRAITS>::GetDependentTrac
         {
             gc.dependentTracker = (LAHASHDEPENDENTHASHTRACKERREF)AllocateObject(MscorlibBinder::GetExistingClass(CLASS__LAHASHDEPENDENTHASHTRACKER));
             gc.GCHeapHashForKeyToValueStore = (GCHEAPHASHOBJECTREF)AllocateObject(MscorlibBinder::GetExistingClass(CLASS__GCHEAPHASH));
-            OBJECTHANDLE dependentHandle = GetAppDomain()->CreateDependentHandle(pLoaderAllocator->GetExposedObject(), gc.GCHeapHashForKeyToValueStore);
+
+            OBJECTREF exposedObject = pLoaderAllocator->GetExposedObject();
+            if (exposedObject == NULL)
+            {
+                if (m_globalDependentTrackerRootHandle == NULL)
+                {
+                    // Global LoaderAllocator does not have an exposed object, so create a fake one
+                    exposedObject = AllocateObject(MscorlibBinder::GetExistingClass(CLASS__OBJECT));
+                    m_globalDependentTrackerRootHandle = GetAppDomain()->CreateHandle(exposedObject);
```

Comment:
You need to clean this handle up. See uses of RegisterHandleForCleanup where we create the other handles. Otherwise, you'll have created a memory leak for collectible assemblies.

Reply:
I added the RegisterHandleForCleanup and am going to check in since it's the only thing you mentioned. If you have further feedback let me know tomorrow and I'll address it in a new PR.